var/spack/repos/builtin/packages/papi/package.py

```diff
@@ -57,6 +91,10 @@ def install(self, spec, prefix):
             # fail, so that PAPI does not get confused
             configure_args.append('MPICC=:')
 
+            configure_args.append('--with-components={0}'.format(
+                ' '.join(spec.variants['components'].value))
```

Comment:
Does this work even if the default `''` is used? You would effectively be passing `--with-components=` to the `configure` invocation.